tests/next_tests/integration_tests/feature_tests/ffront_tests/test_compiled_program.py

```diff
@@ -812,3 +815,45 @@ def test_compile_variants_tuple(cartesian_case, compile_variants_testee_tuple):
         offset_provider=cartesian_case.offset_provider,
     )
     assert np.allclose(out.asnumpy(), field_a.asnumpy() * 3 + field_b.asnumpy() * 4)
+
+
+def test_synchronous_compilation(cartesian_case, compile_testee):
+    # This test is not perfect: only tests that compilation works if '_async_compilation_pool' is not initialized.
+    with mock.patch.object(compiled_program, "_async_compilation_pool", None):
+        a = cases.allocate(cartesian_case, compile_testee, "a")()
+        b = cases.allocate(cartesian_case, compile_testee, "b")()
+
+        out = cases.allocate(cartesian_case, compile_testee, "out")()
+        compile_testee(
+            a,
+            b,
+            out=out,
+            offset_provider=cartesian_case.offset_provider,
+        )
+        assert np.allclose(out.ndarray, a.ndarray + b.ndarray)
+
+
+@pytest.mark.parametrize("synchronous", [True, False], ids=["synchronous", "asynchronous"])
+def test_wait_for_compilation(cartesian_case, compile_testee, synchronous):
+    if cartesian_case.backend is None:
+        pytest.skip("Embedded compiled program doesn't make sense.")
+
+    with (
+        mock.patch.object(compiled_program, "_async_compilation_pool", None)
+        if synchronous
+        else contextlib.nullcontext()
+    ):
+        start_compile = time.time()
+        compile_testee.compile(offset_provider=cartesian_case.offset_provider)
+        end_compile = time.time()
+        compile_call_time = end_compile - start_compile
+
+        start_wait = time.time()
+        gtx.wait_for_compilation()
+        end_wait = time.time()
+        wait_time = end_wait - start_wait
+
+        if synchronous:
+            assert compile_call_time > wait_time
+        else:
+            assert wait_time > compile_call_time
```

Comment:
Are you sure `wait_time > compile_call_time` will always be true for any backend in the asynchronous case? Anyway, it's worth explaining this logic quickly with a comment

Reply:
I think so (note embedded is excluded) as there is always some passes being run and `compile` should return quickly. Of course we consider hash computations in compile as non-critical and this could break in the future... I added a comment.